storage/interfaces.go

```diff
@@ -16,11 +16,30 @@
 
 package storage
 
+import "github.com/Peripli/service-manager/types"
+
 // Storage interface provides entity-specific storages.
 type Storage interface {
 	// Open initializes the storage, e.g. opens a connection to the underlying storage
-	Open() error
+	Open(uri string) error
 
 	// Close clears resources associated with this storage, e.g. closes the connection the underlying storage
 	Close() error
+
+	// Broker provides access to service broker db operations
+	Broker() Broker
+}
+
+type Config interface {
+	createFunc() Storage
+}
+
+// Broker interface for Broker db operations
+//go:generate counterfeiter . Broker
+type Broker interface {
+	Create(broker *types.Broker) error
```

Comment:
these should probably take a context

Reply:
agreed, I was under the impression that sqlx doesn't support context but then I found https://github.com/jmoiron/sqlx/pull/270/files